File: src/main/java/BruteForce/BOJ_3040/Enhanced.java
package bruteforce.boj_3040;

import java.io.*;

// 14120kb 104ms
public class Enhanced {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int[] nums = new int[9];
        boolean flag = false;
        int sum = 0;
        for(int i=0;i<9;i++) {
            nums[i] = Integer.parseInt(br.readLine());
            sum += nums[i];
        }
        for (int i=0; i<9; i++) {
            for (int j=i+1; j<9; j++) {
                if ((sum - (nums[i] + nums[j])) == 100) {
                    for(int n=0;n<9;n++) {
                        if(n != i && n != j) {
                            System.out.println(nums[n]);
                        }
                    }
                    flag = true;
                    break;
                }
            }
            if(flag) break;
        }
        br.close();
    }
}

File: src/main/java/BruteForce/boj_1436/Enhanced.kt
package bruteforce.boj_1436

import java.io.BufferedReader
import java.io.InputStreamReader

// 14596kb 140ms
fun main () {
	val br = BufferedReader(InputStreamReader(System.`in`))
	val r = br.readLine().toInt()
	val arr = arrayListOf<Int>()
	var i = 666
	while (true) {
		if (hasThreeSix(i)) {
			arr.add(i)
		}
		if (arr.size == r) {
			println(arr.last())
			break
		}
		i++
	}
}

fun hasThreeSix(n: Int):Boolean {
	var num = n
	while(num > 0) {
		if (num < 666) return false
		if (num % 1000 == 666) {
			return true
		}
		num /= 10
	}
	return false
}